### ISSUE TITLE
Container termination events lost in 1.6.0

### ISSUE THREAD

**Issue by burke** [docker/docker#11764]
### Summary:

After upgrading to docker 1.6.0, containers cannot be destroyed. The containers run as normal, but the parent process never perceives the container as exited. For example, if I run `docker run --entrypoint=/bin/echo busybox test`, it prints `test` and then hangs indefinitely.

Attempting to `kill`, `stop`, or `rm -f` a container results in:

```
Error response from daemon: Cannot kill container 52e: [2] Container does not exist: container destroyed
FATA[0000] Error: failed to kill one or more containers
```

This only happens on one of the two servers I've upgraded to `1.6.0-rc2` (both of which have the same output for the "System" section below). It happens consistently with any combination of parameters to `docker run`. This was also the behaviour with `1.6.0-rc1`.

I've attached `docker.log` captures for the whole process and attached them to this issue.

I can capture `strace`s or whatever other instrumentation seems helpful, just ask if you need more, or have some pointers on how I might fix this myself.

### System:

```
$ uname -r
3.18.8-031808-generic
$ docker --version
Docker version 1.6.0-rc2, build c5ee149
$ cat /etc/lsb-release | grep DESC
DISTRIB_DESCRIPTION="Ubuntu 14.04 LTS"
```

### Docker logs

1) start docker

```
INFO[0000] +job serveapi(unix:///var/run/docker.sock)
INFO[0000] Listening for HTTP on unix (/var/run/docker.sock)
INFO[0000] +job init_networkdriver()
INFO[0000] -job init_networkdriver() = OK (0)
WARN[0000] mountpoint for memory not found
INFO[0000] Loading containers: start.
................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
INFO[0000] Loading containers: done.
INFO[0000] docker daemon: 1.6.0-rc2 c5ee149; execdriver: native-0.2; graphdriver: overlay
INFO[0000] +job acceptconnections()
INFO[0000] -job acceptconnections() = OK (0)
INFO[0000] Daemon has completed initialization
INFO[0002] GET /v1.18/containers/json
INFO[0002] +job containers()
INFO[0002] -job containers() = OK (0)
INFO[0004] GET /v1.18/containers/json
INFO[0004] +job containers()
INFO[0004] -job containers() = OK (0)
INFO[0009] GET /v1.18/containers/json
INFO[0009] +job containers()
INFO[0009] -job containers() = OK (0)
INFO[0013] GET /v1.18/containers/json
INFO[0013] +job containers()
INFO[0014] -job containers() = OK (0)
INFO[0014] GET /v1.18/containers/json
INFO[0014] +job containers()
INFO[0014] -job containers() = OK (0)
```

2) `docker run --entrypoint=/bin/echo busybox test`

```
INFO[0014] POST /v1.18/containers/create
INFO[0014] +job create()
INFO[0014] +job log(create, 52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8, busybox:latest)
INFO[0014] -job log(create, 52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8, busybox:latest) = OK (0)
INFO[0014] -job create() = OK (0)
INFO[0014] POST /v1.18/containers/52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8/attach?stderr=1&stdout=1&stream=1
INFO[0014] +job container_inspect(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8)
INFO[0014] -job container_inspect(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8) = OK (0)
INFO[0014] +job attach(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8)
INFO[0014] POST /v1.18/containers/52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8/start
INFO[0014] +job start(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8)
INFO[0014] +job allocate_interface(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8)
INFO[0014] -job allocate_interface(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8) = OK (0)
INFO[0014] +job log(start, 52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8, busybox:latest)
INFO[0014] -job log(start, 52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8, busybox:latest) = OK (0)
INFO[0014] GET /images/json?all=False
INFO[0014] +job images()
INFO[0014] -job images() = OK (0)
INFO[0014] GET /images/json?all=True
INFO[0014] +job images()
INFO[0014] -job images() = OK (0)
INFO[0015] GET /containers/json?all=False&size=False
INFO[0015] +job containers()
INFO[0015] -job start(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8) = OK (0)
INFO[0015] -job containers() = OK (0)
WARN[0015] Your kernel does not support OOM notifications: There is no path for "memory" in state
INFO[0015] GET /containers/json?all=True&size=False
INFO[0015] +job containers()
INFO[0015] -job containers() = OK (0)
INFO[0015] GET /events?since=1427317521&until=1427317579
INFO[0015] +job events()
INFO[0015] -job events() = OK (0)
INFO[0018] GET /v1.18/containers/json
INFO[0018] +job containers()
INFO[0018] -job containers() = OK (0)
INFO[0020] GET /v1.18/containers/json
INFO[0020] +job containers()
INFO[0020] -job containers() = OK (0)
INFO[0022] GET /v1.18/containers/json
INFO[0022] +job containers()
INFO[0022] -job containers() = OK (0)
INFO[0025] GET /v1.18/containers/json
INFO[0025] +job containers()
INFO[0025] -job containers() = OK (0)
```

3) Notice that container hasn't terminated yet. Hit `^C` in shell running container.

```
INFO[0026] POST /v1.18/containers/52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8/kill?signal=INT
INFO[0026] +job kill(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8, INT)
Cannot kill container 52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8: [2] Container does not exist: container destroyed
INFO[0026] -job kill(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8, INT) = ERR (1)
ERRO[0026] Handler for POST /containers/{name:.*}/kill returned error: Cannot kill container 52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8: [2] Container does not exist: container destroyed
ERRO[0026] HTTP Error: statusCode=500 Cannot kill container 52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8: [2] Container does not exist: container destroyed
INFO[0029] GET /v1.18/containers/json
INFO[0029] +job containers()
INFO[0029] -job containers() = OK (0)
INFO[0032] GET /v1.18/containers/json
INFO[0032] +job containers()
INFO[0032] -job containers() = OK (0)
INFO[0033] POST /v1.18/containers/52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8/kill?signal=HUP
INFO[0033] +job kill(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8, HUP)
Cannot kill container 52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8: [2] Container does not exist: container destroyed
INFO[0033] -job kill(52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8, HUP) = ERR (1)
ERRO[0033] Handler for POST /containers/{name:.*}/kill returned error: Cannot kill container 52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8: [2] Container does not exist: container destroyed
ERRO[0033] HTTP Error: statusCode=500 Cannot kill container 52e73e5961d5d5a76a5a70e9a9e253af6dc1b5cfa95e8623cc820f46eae52ca8: [2] Container does not exist: container destroyed
INFO[0033] GET /v1.18/containers/json
INFO[0033] +job containers()
INFO[0033] -job containers() = OK (0)
INFO[0034] GET /images/json?all=False
INFO[0034] +job images()
INFO[0034] -job images() = OK (0)
INFO[0034] GET /images/json?all=True
INFO[0034] +job images()
INFO[0034] -job images() = OK (0)
INFO[0034] GET /containers/json?all=False&size=False
INFO[0034] +job containers()
INFO[0034] -job containers() = OK (0)
INFO[0034] GET /containers/json?all=True&size=False
INFO[0034] +job containers()
INFO[0034] -job containers() = OK (0)
INFO[0034] GET /events?since=1427317579&until=1427317598
INFO[0034] +job events()
INFO[0034] -job events() = OK (0)
INFO[0036] GET /v1.18/containers/json
INFO[0036] +job containers()
INFO[0036] -job containers() = OK (0)
```

**Comment by burke** [comment]
Possibly interesting:

* This happens even if I start docker with `docker -d --graph=$(mktemp -d)` (ie. no history)

* This happens even if I don't use `--storage-driver=overlay` (looks like it chose `devicemapper`)

**Comment by burke** [comment]
Actually, I take back that thing about kernel homogeneity -- the machine it's working on is `3.18.6-031806-generic`, so I'd be inclined to think it's a problem with `3.18.8`.

**Comment by Piste** [comment]
+kind/regression

**Comment by burke** [comment]
Here's a strace taken of the daemon: https://gist.github.com/burke/bcfaea4bf0dd17805a38

1. stop docker
2. nuke docker root
3. start docker
4. pull busybox image
5. restart docker
6. start strace on docker daemon with `-tt -T -f`
7. `docker run busybox echo`
8. wait a couple seconds and stop strace.

I've trimmed the trace to begin where "busybox" is looked, and to end shortly after the daemon returned to what appeared to be steady-state.

**Comment by LK4D4** [comment]
@burke Main problem that I can't reproduce this :/ Is there something special about your upgrade process, could you provide line with which you run docker?

**Comment by LK4D4** [comment]
@burke Sorry, I reproduced :)

**Comment by LK4D4** [comment]
This is definetely fixed in my replace_execdriver branch, though I have no idea how.

**Comment by mbentley** [comment]
I am also able to reproduce this behavior on Debian sid when running nginx with `docker run -P nginx:latest` and then doing a ctrl-c to stop the container.  The nginx process is no longer running but docker seems to think the container is still running.

```
$ uname -a
Linux localdev 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt7-1 (2015-03-01) x86_64 GNU/Linux
$ docker info
Containers: 4
Images: 632
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 640
Execution Driver: native-0.2
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
CPUs: 8
Total Memory: 1.963 GiB
Name: localdev
ID: ABO2:4REM:BS7R:OXYG:R7V6:UJGL:2YTX:CJ7H:UEYY:IK7O:JXYQ:YBYX
Username: mbentley
Registry: [https://index.docker.io/v1/]
WARNING: No memory limit support
WARNING: No swap limit support
```

I installed docker by having the `lxc-docker-1.5.0` package from get.docker.com installed and dropping in the new binary for `1.6.0-rc2`.

Output from the daemon when doing a ctrl-c:
```
Mar 30 10:11:31 localdev docker[16652]: time="2015-03-30T10:11:31-04:00" level=info msg="POST /v1.18/containers/480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec/kill?signal=WINCH"
Mar 30 10:11:31 localdev docker[16652]: time="2015-03-30T10:11:31-04:00" level=info msg="+job kill(480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec, WINCH)"
Mar 30 10:11:31 localdev docker[16652]: Cannot kill container 480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec: [2] Container does not exist: container destroyed
Mar 30 10:11:31 localdev docker[16652]: time="2015-03-30T10:11:31-04:00" level=info msg="-job kill(480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec, WINCH) = ERR (1)"
Mar 30 10:11:31 localdev docker[16652]: time="2015-03-30T10:11:31-04:00" level=error msg="Handler for POST /containers/{name:.*}/kill returned error: Cannot kill container 480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec: [2] Container does not exist: container destroyed"
Mar 30 10:11:31 localdev docker[16652]: time="2015-03-30T10:11:31-04:00" level=error msg="HTTP Error: statusCode=500 Cannot kill container 480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec: [2] Container does not exist: container destroyed"
Mar 30 10:11:33 localdev docker[16652]: time="2015-03-30T10:11:33-04:00" level=info msg="POST /v1.18/containers/480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec/kill?signal=INT"
Mar 30 10:11:33 localdev docker[16652]: time="2015-03-30T10:11:33-04:00" level=info msg="+job kill(480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec, INT)"
Mar 30 10:11:33 localdev docker[16652]: Cannot kill container 480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec: [2] Container does not exist: container destroyed
Mar 30 10:11:33 localdev docker[16652]: time="2015-03-30T10:11:33-04:00" level=info msg="-job kill(480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec, INT) = ERR (1)"
Mar 30 10:11:33 localdev docker[16652]: time="2015-03-30T10:11:33-04:00" level=error msg="Handler for POST /containers/{name:.*}/kill returned error: Cannot kill container 480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec: [2] Container does not exist: container destroyed"
Mar 30 10:11:33 localdev docker[16652]: time="2015-03-30T10:11:33-04:00" level=error msg="HTTP Error: statusCode=500 Cannot kill container 480251af2e43c2c22b4cf6eb75c451b149315a5cdf1313096a21a183df8620ec: [2] Container does not exist: container destroyed"
```

I also tried to start a container using the same command as above and then tried to do a `docker kill` on the container and I only see:
```
Mar 30 10:24:33 localdev docker[20269]: time="2015-03-30T10:24:33-04:00" level=info msg="POST /v1.18/containers/4ad/kill?signal=KILL"
Mar 30 10:24:33 localdev docker[20269]: time="2015-03-30T10:24:33-04:00" level=info msg="+job kill(4ad, KILL)"
Mar 30 10:24:43 localdev docker[20269]: time="2015-03-30T10:24:43-04:00" level=info msg="Container 4adaaa596cfe failed to exit within 10 seconds of kill - trying direct SIGKILL"
```

The `docker kill` command just hangs with no output.  Then when I try to run the `docker kill` again, I get
```
$ docker kill 4ad
Error response from daemon: Cannot kill container 4ad: [2] Container does not exist: container destroyed
FATA[0000] Error: failed to kill one or more containers
```

**Comment by burke** [comment]
This is still broken for me on RC3:

```
burke@sb3.chi:~ $ docker version
Client version: 1.6.0-rc3
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 20d4e6f
OS/Arch (client): linux/amd64
Server version: 1.6.0-rc3
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 20d4e6f
OS/Arch (server): linux/amd64
burke@sb3.chi:~ $ docker run --entrypoint=/bin/echo busybox test
test
# (doesn't terminate)
```

**Comment by LK4D4** [comment]
@burke weird :( in rc2 I can reproduce this easy, but now:
```
while docker run --entrypoint=/bin/echo busybox test; do :; done
```
working until my machine is out of fds.

**Comment by mbentley** [comment]
I can reproduce as well:
```
$ docker run --entrypoint=/bin/echo busybox test
test
^C
```

and I can't kill the container:
```
$ docker kill 0d3
Error response from daemon: Cannot kill container 0d3: [2] Container does not exist: container destroyed
FATA[0000] Error: failed to kill one or more containers
```

```
$ docker info
Containers: 1
Images: 654
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 656
 Dirperm1 Supported: true
Execution Driver: native-0.2
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
CPUs: 8
Total Memory: 1.963 GiB
Name: localdev
ID: ABO2:4REM:BS7R:OXYG:R7V6:UJGL:2YTX:CJ7H:UEYY:IK7O:JXYQ:YBYX
Username: mbentley
Registry: [https://index.docker.io/v1/]
WARNING: No memory limit support
WARNING: No swap limit support
```

**Comment by burke** [comment]
Not much helpful in the log, even with `-l debug`:

```
DEBU[0029] Calling POST /containers/create
INFO[0029] POST /v1.18/containers/create
INFO[0029] +job create()
INFO[0029] +job log(create, 272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09, busybox:latest)
INFO[0029] -job log(create, 272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09, busybox:latest) = OK (0)
INFO[0029] -job create() = OK (0)
DEBU[0029] Calling POST /containers/{name:.*}/attach
INFO[0029] POST /v1.18/containers/272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09/attach?stderr=1&stdout=1&stream=1
INFO[0029] +job container_inspect(272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09)
INFO[0029] -job container_inspect(272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09) = OK (0)
INFO[0029] +job attach(272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09)
DEBU[0029] attach: stdout: begin
DEBU[0029] attach: stderr: begin
DEBU[0029] Calling POST /containers/{name:.*}/start
INFO[0029] POST /v1.18/containers/272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09/start
INFO[0029] +job start(272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09)
INFO[0029] +job allocate_interface(272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09)
INFO[0029] -job allocate_interface(272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09) = OK (0)
INFO[0029] +job log(start, 272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09, busybox:latest)
INFO[0029] -job log(start, 272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09, busybox:latest) = OK (0)
INFO[0029] -job start(272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09) = OK (0)
WARN[0029] Your kernel does not support OOM notifications: There is no path for "memory" in state
DEBU[0032] Calling GET /containers/json
INFO[0032] GET /v1.18/containers/json
INFO[0032] +job containers()
INFO[0032] -job containers() = OK (0)
DEBU[0033] Calling GET /containers/json
INFO[0033] GET /v1.18/containers/json
INFO[0033] +job containers()
INFO[0033] -job containers() = OK (0)
DEBU[0034] Calling POST /containers/{name:.*}/kill
INFO[0034] POST /v1.18/containers/272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09/kill?signal=INT
INFO[0034] +job kill(272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09, INT)
DEBU[0034] Sending 2 to 272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09
Cannot kill container 272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09: [2] Container does not exist: container destroyed
INFO[0034] -job kill(272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09, INT) = ERR (1)
ERRO[0034] Handler for POST /containers/{name:.*}/kill returned error: Cannot kill container 272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09: [2] Container does not exist: container destroyed
ERRO[0034] HTTP Error: statusCode=500 Cannot kill container 272bc4658fe1cfd47356a673acc2c4a0b1847153d7d3abd74c1387d00c860a09: [2] Container does not exist: container destroyed
```

**Comment by LK4D4** [comment]
@jfrazelle Can you reproduce on your machine?

**Comment by jfrazelle** [comment]
I can't reproduce :(

**Comment by jfrazelle** [comment]
Could it be a kernel level issue?

**Comment by burke** [comment]
I assumed it was a 3.18.8 thing, but @mbentley is seeing it with 3.16.0 apparently.

**Comment by crosbymichael** [comment]
i'm checkin' this out, i'll let you know what I find

**Comment by crosbymichael** [comment]
Found it', i'll have a fix shortly

**Comment by crosbymichael** [comment]
The fix for this is linked to the PR.  Thanks for the report and logs

**Comment by donhcd** [comment]
@crosbymichael  I'm still seeing this in rc5:

```
$ docker kill 959addc7ab30
Error response from daemon: Cannot kill container 959addc7ab30: [2] Container does not exist: container destroyed
FATA[0000] Error: failed to kill one or more containers
$ docker inspect 959addc7ab
[{
    "AppArmorProfile": "",
    "Args": [],
    "Config": {
        "AttachStderr": false,
        "AttachStdin": false,
        "AttachStdout": false,
        "Cmd": [],
        "CpuShares": 0,
        "Cpuset": "",
        "Domainname": "",
        "Entrypoint": [
            "server"
        ],
        "Env": [
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/go/bin:/var/cache/drone/bin",
            "GOLANG_VERSION=1.4rc1",
            "GOPATH=/root/go",
            "GOROOT=/usr/local/go",
            "LANG=C"
        ],
        "ExposedPorts": {
            "8080/tcp": {}
        },
        "Hostname": "959addc7ab30",
        "Image": "a5a2af8a555e",
        "Labels": {},
        "MacAddress": "",
        "Memory": 0,
        "MemorySwap": 0,
        "NetworkDisabled": false,
        "OnBuild": null,
        "OpenStdin": false,
        "PortSpecs": null,
        "StdinOnce": false,
        "Tty": false,
        "User": "",
        "Volumes": null,
        "WorkingDir": ""
    },
    "Created": "2015-04-09T18:42:58.769851367Z",
    "Driver": "aufs",
    "ExecDriver": "native-0.2",
    "ExecIDs": null,
    "HostConfig": {
        "Binds": [
            "/usr/local/containerVol:/etc/vol:rw",
            "/var/local/databases:/databases:rw"
        ],
        "CapAdd": null,
        "CapDrop": null,
        "CgroupParent": "",
        "ContainerIDFile": "",
        "CpuShares": 0,
        "CpusetCpus": "",
        "Devices": null,
        "Dns": null,
        "DnsSearch": null,
        "ExtraHosts": null,
        "IpcMode": "",
        "Links": null,
        "LogConfig": {
            "Config": null,
            "Type": "json-file"
        },
        "LxcConf": null,
        "Memory": 0,
        "MemorySwap": 0,
        "NetworkMode": "",
        "PidMode": "",
        "PortBindings": {},
        "Privileged": false,
        "PublishAllPorts": false,
        "ReadonlyRootfs": false,
        "RestartPolicy": {
            "MaximumRetryCount": 0,
            "Name": "always"
        },
        "SecurityOpt": null,
        "Ulimits": null,
        "VolumesFrom": null
    },
    "HostnamePath": "/var/lib/docker/containers/959addc7ab30144bacd2f6521c70313fe34afb6ec11c6c4a7c7a0844cfff2bd5/hostname",
    "HostsPath": "/var/lib/docker/containers/959addc7ab30144bacd2f6521c70313fe34afb6ec11c6c4a7c7a0844cfff2bd5/hosts",
    "Id": "959addc7ab30144bacd2f6521c70313fe34afb6ec11c6c4a7c7a0844cfff2bd5",
    "Image": "8840c8cf3be0072412c07edba8cb147b833d6cc626df93464d03b7f2b90e229f",
    "LogPath": "",
    "MountLabel": "",
    "Name": "foobar",
    "NetworkSettings": {
        "Bridge": "docker0",
        "Gateway": "172.17.42.1",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "IPAddress": "172.17.0.1",
        "IPPrefixLen": 16,
        "IPv6Gateway": "",
        "LinkLocalIPv6Address": "fe80::42:acff:fe11:1",
        "LinkLocalIPv6PrefixLen": 64,
        "MacAddress": "02:42:ac:11:00:01",
        "PortMapping": null,
        "Ports": {
            "8080/tcp": null
        }
    },
    "Path": "server",
    "ProcessLabel": "",
    "ResolvConfPath": "/var/lib/docker/containers/959addc7ab30144bacd2f6521c70313fe34afb6ec11c6c4a7c7a0844cfff2bd5/resolv.conf",
    "RestartCount": 0,
    "State": {
        "Dead": false,
        "Error": "",
        "ExitCode": 0,
        "FinishedAt": "2015-04-10T00:17:17.387730447Z",
        "OOMKilled": false,
        "Paused": false,
        "Pid": 10721,
        "Restarting": false,
        "Running": true,
        "StartedAt": "2015-04-10T00:17:17.519076764Z"
    },
    "Volumes": {
        "/databases": "/var/local/databases",
        "/etc/vol": "/usr/local/containerVol"
    },
    "VolumesRW": {
        "/databases": true,
        "/etc/vol": true
    }
}
]
$ ps 10721
  PID TTY      STAT   TIME COMMAND
$
```

System Info
=========
```
$ uname -r
3.13.0-36-generic
$ docker version
Client API version: 1.16
Go version (client): go1.3.3
OS/Arch (client): linux/amd64
Server version: 1.6.0-rc5
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): fc4825d
$ cat /etc/lsb-release | grep DESC
DISTRIB_DESCRIPTION="Ubuntu 14.04.1 LTS"
```

**Comment by jfrazelle** [comment]
`docker info` too

**Comment by jfrazelle** [comment]
The id you are killing and inspecting are not the same

**Comment by donhcd** [comment]
```$ docker info
Containers: 26
Images: 624
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 28960
 Dirperm1 Supported: false
Execution Driver: native-0.2
Kernel Version: 3.13.0-36-generic
Operating System: Ubuntu 14.04.1 LTS
CPUs: 1
Total Memory: 1.955 GiB
Name: vagrant-ubuntu-trusty-64
ID: NUJZ:42WR:PCKJ:7XR2:VBGZ:PU44:C4ML:DIAQ:KP64:FV7E:TK6Y:MWP5
Debug mode (server): true
Debug mode (client): false
Fds: 26
Goroutines: 52
EventsListeners: 0
Init SHA1: b70e69f43ba7665d9c66511a4a1d7b0266c4c756
Init Path: /usr/bin/docker
Docker Root Dir: /var/lib/docker
Username: donhcd
Registry: [https://index.docker.io/v1/]
WARNING: No swap limit support```

sorry, I got lazy typing the inspect, but the two IDs are equivalent:
```
 % docker inspect -f '{{.Id}}' 959addc7ab30
959addc7ab30144bacd2f6521c70313fe34afb6ec11c6c4a7c7a0844cfff2bd5
 % docker inspect -f '{{.Id}}' 959addc7ab
959addc7ab30144bacd2f6521c70313fe34afb6ec11c6c4a7c7a0844cfff2bd5
```

**Comment by jfrazelle** [comment]
@donhcd I reopened. We need exact steps you used to reproduce.

**Comment by jfrazelle** [comment]
is it with `docker run --entrypoint=/bin/echo busybox test` I wonder what is the output of `lsmod`

**Comment by donhcd** [comment]
```
$ curl https://raw.githubusercontent.com/docker/docker/master/contrib/check-config.sh|bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4877  100  4877    0     0    830      0  0:00:05  0:00:05 --:--:--  1121
warning: /proc/config.gz does not exist, searching other paths for kernel config...
info: reading kernel config from /boot/config-3.13.0-36-generic ...

Generally Necessary:
- cgroup hierarchy: properly mounted [/sys/fs/cgroup]
- apparmor: enabled and tools installed
- CONFIG_NAMESPACES: enabled
- CONFIG_NET_NS: enabled
- CONFIG_PID_NS: enabled
- CONFIG_IPC_NS: enabled
- CONFIG_UTS_NS: enabled
- CONFIG_DEVPTS_MULTIPLE_INSTANCES: enabled
- CONFIG_CGROUPS: enabled
- CONFIG_CGROUP_CPUACCT: enabled
- CONFIG_CGROUP_DEVICE: enabled
- CONFIG_CGROUP_FREEZER: enabled
- CONFIG_CGROUP_SCHED: enabled
- CONFIG_CPUSETS: enabled
- CONFIG_MACVLAN: enabled
- CONFIG_VETH: enabled
- CONFIG_BRIDGE: enabled
- CONFIG_NF_NAT_IPV4: enabled
- CONFIG_IP_NF_FILTER: enabled
- CONFIG_IP_NF_TARGET_MASQUERADE: enabled
- CONFIG_NETFILTER_XT_MATCH_ADDRTYPE: enabled
- CONFIG_NETFILTER_XT_MATCH_CONNTRACK: enabled
- CONFIG_NF_NAT: enabled
- CONFIG_NF_NAT_NEEDED: enabled
- CONFIG_POSIX_MQUEUE: enabled

Optional Features:
- CONFIG_MEMCG_SWAP: enabled
- CONFIG_MEMCG_SWAP_ENABLED: missing
    (note that cgroup swap accounting is not enabled in your kernel config, you can enable it by setting boot option "swapaccount=1")
- CONFIG_RESOURCE_COUNTERS: enabled
- CONFIG_CGROUP_PERF: enabled
- Storage Drivers:
  - "aufs":
    - CONFIG_AUFS_FS: enabled
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
    - CONFIG_EXT4_FS_SECURITY: enabled
  - "btrfs":
    - CONFIG_BTRFS_FS: enabled
  - "devicemapper":
    - CONFIG_BLK_DEV_DM: enabled
    - CONFIG_DM_THIN_PROVISIONING: enabled
    - CONFIG_EXT4_FS: enabled
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
    - CONFIG_EXT4_FS_SECURITY: enabled
  - "overlay":
    - CONFIG_OVERLAY_FS: missing
    - CONFIG_EXT4_FS_SECURITY: enabled
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
```

**Comment by donhcd** [comment]
@crosbymichael should this be closed due to #12423 ?
